rllib/algorithms/ppo/tests/test_ppo_rl_module.py

```diff
@@ -105,126 +207,151 @@ def setUpClass(cls):
     def tearDownClass(cls):
         ray.shutdown()
 
+    def get_ppo_module(self, framwework, env, lstm, shared_encoder):
+        if framwework == "torch":
+            config = get_expected_model_config_torch(env, lstm, shared_encoder)
+            module = PPOTorchRLModule(config)
+        else:
+            config = get_expected_model_config_tf(env, shared_encoder)
+            module = PPOTfModule(config)
+        return module
+
+    def get_input_batch_from_obs(self, framework, obs):
+        if framework == "torch":
+            batch = {
+                SampleBatch.OBS: convert_to_torch_tensor(obs)[None],
+            }
+        else:
+            batch = {SampleBatch.OBS: np.array([obs])}
+        return batch
+
     def test_rollouts(self):
         # TODO: Add BreakoutNoFrameskip-v4 to cover a 3D obs space
-        for env_name in ["CartPole-v1", "Pendulum-v1"]:
-            for fwd_fn in ["forward_exploration", "forward_inference"]:
+        for fw in framework_iterator(frameworks=["torch", "tf2"]):
+            for env_name in ["CartPole-v1", "Pendulum-v1"]:
+                for fwd_fn in ["forward_exploration", "forward_inference"]:
+                    for shared_encoder in [False, True]:
+                        for lstm in [True, False]:
```

Comment:
these for loops look ... you know
can you use itertools instead?
https://stackoverflow.com/questions/798854/all-combinations-of-a-list-of-lists